controllers/utils/base_reconciler.go

```diff
@@ -349,18 +369,24 @@ func (r *ReconciliationRunner) CheckIfPaused() (ctrl.Result, error) {
 	return reconcile.Result{}, nil
 }
 
+// SetReturnEarly sets the runner to return early. This causes the runner to break from running further
+// reconciliation stages and return whatever result the current method returns.
+func (r *ReconciliationRunner) SetReturnEarly() {
+	r.returnEarly = true
+}
+
 // GetReconcilationSubject gets the reconciliation subject of type defined by the concrete reconciler. It also sets up
 // a patch helper at this point.
 func (r *ReconciliationRunner) GetReconciliationSubject() (res ctrl.Result, reterr error) {
 	r.Log.V(1).Info("Getting reconciliation subject.")
 	err := client.IgnoreNotFound(r.K8sClient.Get(r.RequestCtx, r.Request.NamespacedName, r.ReconciliationSubject))
-	if r.ReconciliationSubject.GetName() == "" {
+	if r.ReconciliationSubject.GetName() == "" { // Resource does not exist. No need to reconcile.
```

Comment:
will this scenario be logged? It may be helpful when troubleshooting

Reply:
No, and while the coding may look different the logic hasn't changed. We've not been logging it to date, but I'll toss in a verbosity level 1 info log.